dataplicity/m2m/packetbase.py

```diff
@@ -31,6 +31,9 @@ def __new__(mcs, name, bases, attrs):
             if packet_cls.type >= 0:
                 assert packet_cls.type not in packet_cls.registry, "packet type {!r} has already been registered".format(packet_cls, type)
                 packet_cls.registry[packet_cls.type] = packet_cls
+                # register the packet type in base class so we can have a
+                # generic interface of accessing it.
+                bases[0].registry[packet_cls.type] = packet_cls
```

Comment:
Before this line, I could only construct the class by verbosely specifying it's name, i.e.

`PingPacket.from_bytes(b' ... ')`

but this isn't really helpful if I have just *a* binary packet and I want to check what packet it is.

Also, the docstring for the PacketMeta states:

`"""Maintains a registry of packet classes."""`

But this is not the case - the .registry attribute would only be visible within this particular class